File: Ex01-Intro/thread_intro.py
from threading import Thread

i = 0

def thread_func1():
	global i
	for j in range(0, 1000000):
		i += 1

def thread_func2():
	global i
	for j in range(0, 1000000):
		i -= 1

def main():
	print("Before thread join: " + str(i))

	thread_1 = Thread(target = thread_func1, args = (),)
	thread_2 = Thread(target = thread_func2, args = (),)
	thread_1.start()
	thread_2.start()

	thread_1.join()
	thread_2.join()
	
	print("After thread join: " + str(i))

main()

File: Ex01-Intro/C_code/thread_intro.c
//haakoneh & erlendvd

#include <pthread.h>
#include <stdio.h>


int i = 0;

void* thread_func_1(){
	unsigned int j;
	for(j = 0; j < 1000000; j++){
		i++;
	}
	return NULL;
}

void* thread_func_2(){
	unsigned int j;
	for(j = 0; j < 1000000; j++){
		i--;
	}
	return NULL;
}

int main(){
    pthread_t thread_1;
    pthread_t thread_2;

    pthread_create(&thread_1, NULL, thread_func_1, NULL);
    pthread_create(&thread_2, NULL, thread_func_2, NULL);

    printf("Before joined threads:  %i\n", i);

    pthread_join(thread_1, NULL);
    pthread_join(thread_2, NULL);

    printf("After joined threads:  %i\n", i);

    return 0;
}
